Library/src/Systems/Spaces/SpaceSystem.cpp

```diff
@@ -260,44 +269,50 @@ void SpaceSystem::ExitSpace(NullResultCallback Callback)
 {
 	CSP_LOG_FORMAT(LogLevel::Log, "Exiting Space %s", CurrentSpace.Name.c_str());
 
-    // As the user is exiting the space, we now clear all scopes that they are registered to.
-    auto* MultiplayerConnection = csp::systems::SystemsManager::Get().GetMultiplayerConnection();
-    if(MultiplayerConnection != nullptr)
-    {
+	// As the user is exiting the space, we now clear all scopes that they are registered to.
+	auto& SystemsManager		= systems::SystemsManager::Get();
+	auto* MultiplayerConnection = SystemsManager.GetMultiplayerConnection();
+
+	if (MultiplayerConnection != nullptr)
+	{
+		csp::systems::SystemsManager::Get().GetSpaceEntitySystem()->Shutdown();
+
 		MultiplayerConnection->StopListening(
-		   [MultiplayerConnection, Callback](csp::multiplayer::ErrorCode Error)
-		   {
-				if (Error != csp::multiplayer::ErrorCode::None)
+			[MultiplayerConnection, Callback](multiplayer::ErrorCode Error)
+			{
+				if (Error != multiplayer::ErrorCode::None)
 				{
-					CSP_LOG_ERROR_FORMAT("Error on exiting spaces, whilst stopping listening in order to clear scopes, ErrorCode: %s", csp::multiplayer::ErrorCodeToString(Error).c_str());
-                    INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
+					CSP_LOG_ERROR_FORMAT("Error on exiting spaces, whilst stopping listening in order to clear scopes, ErrorCode: %s",
+										 multiplayer::ErrorCodeToString(Error).c_str());
+					INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
+
+					return;
 				}
-				else
-				{
-					MultiplayerConnection->ResetScopes([Callback](csp::multiplayer::ErrorCode Error)
+
+				MultiplayerConnection->ResetScopes(
+					[Callback](multiplayer::ErrorCode Error)
 					{
-						csp::systems::SystemsManager::Get().GetSpaceEntitySystem()->LocalDestroyAllEntities();
-						if (Error != csp::multiplayer::ErrorCode::None)
+						if (Error != multiplayer::ErrorCode::None)
 						{
-						    CSP_LOG_ERROR_FORMAT("Error on exiting spaces whilst clearing scopes, ErrorCode: %s", csp::multiplayer::ErrorCodeToString(Error).c_str());
-                            INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
+							CSP_LOG_ERROR_FORMAT("Error on exiting spaces whilst clearing scopes, ErrorCode: %s",
+												 multiplayer::ErrorCodeToString(Error).c_str());
+							INVOKE_IF_NOT_NULL(Callback, MakeInvalid<NullResult>());
+
+							return;
```

Comment:
Nitpick - please could we follow an if/else pattern here rather than earlying out?

Reply:
We specifically don't do that in most of our code. We always early out for errors.

Reply:
That's fair. From a quick scan of `SpaceSystemc.pp` I agree it is used more frequently. Resolving!